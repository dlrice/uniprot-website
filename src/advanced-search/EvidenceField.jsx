import React, { Fragment } from 'react';

// type Props = {
//   updateEvidence: Function,
//   selectedEvidence?: string,
//   url: string,
//   data: Array<Group>,
// };

const EvidenceField = ({ value = '', handleChange, data = [] }) => {
  if (!data) {
    return null;
  }

  return (
    <Fragment>
      <label htmlFor="evidence_select">
        Evidence
        <select id="evidence_select" value={value} onChange={e => handleChange(e.target.value)}>
          {data.map(group => (
            <optgroup label={group.groupName} key={group.groupName}>
              {group.items.map(item => (
                <option value={item.code} key={item.code}>
                  {item.name}
                </option>
              ))}
            </optgroup>
          ))}
        </select>
      </label>
    </Fragment>
  );
};

export default EvidenceField;
