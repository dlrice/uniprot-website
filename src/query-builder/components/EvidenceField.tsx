import React, { useEffect, useState } from 'react';
import { QueryBit, SearchTermType } from '../types/searchTypes';
import initializer from '../utils/fieldInitializer';

const EvidenceField: React.FC<{
  field: SearchTermType;
  handleChange: (queryBit: QueryBit) => void;
  initialValue?: QueryBit;
}> = ({ field, handleChange, initialValue }) => {
  const [value, setValue] = useState(() => initializer(field, initialValue));

  useEffect(() => {
    if (value.length > 0) {
      handleChange({ [field.id]: `(${field.term}:${value?.trim()})` });
    }
  }, [field, value, handleChange]);

  return (
    <label htmlFor="evidence_select">
      Evidence
      <select
        id="evidence_select"
        data-testid="evidence-select"
        value={value}
        onChange={(e) => setValue(e.target.value)}
      >
        {field.evidenceGroups?.map((group) => (
          <optgroup label={group.groupName} key={group.groupName}>
            {group.items.map((item: { code: string; name: string }) => (
              <option value={item.code} key={item.code}>
                {item.name}
              </option>
            ))}
          </optgroup>
        ))}
      </select>
    </label>
  );
};

export default EvidenceField;
