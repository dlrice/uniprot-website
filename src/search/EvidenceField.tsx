import React, { Fragment } from 'react';

type EvidenceFieldProps = {
  value: string | undefined;
  handleChange: (value: string) => void;
  data: any[];
};

const EvidenceField: React.FC<EvidenceFieldProps> = ({ value = '', handleChange, data = [] }) => {
  if (!data) {
    return null;
  }

  return (
    <Fragment>
      <label htmlFor="evidence_select">
        Evidence
        <select id="evidence_select" value={value} onChange={e => handleChange(e.target.value)}>
          {data.map(group => (
            <optgroup label={group.groupName} key={group.groupName}>
              {group.items.map((item: { code: string; name: string }) => (
                <option value={item.code} key={item.code}>
                  {item.name}
                </option>
              ))}
            </optgroup>
          ))}
        </select>
      </label>
    </Fragment>
  );
};

export default EvidenceField;
