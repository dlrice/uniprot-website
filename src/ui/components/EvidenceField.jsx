import React from 'react';

// type Props = {
//   updateEvidence: Function,
//   selectedEvidence?: string,
//   url: string,
//   data: Array<Group>,
// };

const EvidenceField = ({ field, value='', handleChange, data=[] }) => {
  if (!data) {
    return null;
  }

  return (
    <div className="advanced-search__inputs">
      <label htmlFor="evidence_select">
        Evidence
        <select
          id="evidence_select"
          value={value}
          onChange={e => handleChange(e.target.value)}
        >
          {data.map(group => (
            <optgroup label={group.groupName} key={group.groupName}>
              {group.items.map(item => (
                <option value={item.code} key={item.code}>
                  {item.name}
                </option>
              ))}
            </optgroup>
          ))}
        </select>
      </label>
    </div>
  );
}

export default EvidenceField;
