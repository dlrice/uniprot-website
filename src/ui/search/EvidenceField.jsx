// @flow
import React, { Component } from 'react';
import withData from '../hoc/withData';

type Props = {
  updateEvidence: Function,
  selectedEvidence?: string,
  url: string,
  data: Array<Group>,
};

class EvidenceField extends Component<Props> {
  static defaultProps = {
    selectedEvidence: '',
  };

  render() {
    const { data, selectedEvidence, updateEvidence } = this.props;
    if (!data) {
      return null;
    }
    return (
      <div className="advanced-search__inputs">
        <label htmlFor="evidence_select">
          Evidence
          <select id="evidence_select" value={selectedEvidence} onChange={e => updateEvidence(e)}>
            {data.map(group => (
              <optgroup label={group.groupName} key={group.groupName}>
                {group.items.map(item => (
                  <option value={item.code} key={item.code}>
                    {item.name}
                  </option>
                ))}
              </optgroup>
            ))}
          </select>
        </label>
      </div>
    );
  }
}

export default withData(props => props.url)(EvidenceField);
