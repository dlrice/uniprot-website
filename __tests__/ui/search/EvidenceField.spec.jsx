import React from 'react';
import renderer from 'react-test-renderer';

import EvidenceField from '../../../src/ui/search/EvidenceField';

describe('EvidenceField component', () => {
  test('should render', () => {
    const component = renderer.create(<EvidenceField />).toJSON();

    expect(component).toMatchSnapshot();
  });
});
