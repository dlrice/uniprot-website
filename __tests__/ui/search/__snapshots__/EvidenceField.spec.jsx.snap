// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EvidenceField component should render 1`] = `null`;
